template/.github/workflows/weekly_windows_macos.yml

```diff
@@ -0,0 +1,29 @@
+name: Windows and MacOS weekly tests
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * 1'
+
+jobs:
+  tests:
+    name: Tests
+    strategy:
+      matrix:
+        os: ['macos-latest', 'windows-latest']
+        python:
+          - version: '${{needs.formatting.outputs.min_python}}'
+            tox-env: '${{needs.formatting.outputs.min_tox_env}}'
+    uses: ./.github/workflows/test.yml
+    with:
+      os-variant: ${{ matrix.os }}
+      python-version: ${{ matrix.python.version }}
+      tox-env: ${{ matrix.python.tox-env }}
+
+  docs:
+    needs: tests
+    uses: ./.github/workflows/docs.yml
+    with:
+      publish: false
+      linkcheck: ${{ contains(matrix.variant.os, 'ubuntu') && github.ref == 'refs/heads/main' }}
```

Comment:
I would say we don't need to run linkcheck on these builds? Would just add more noise (failed runs even if all tests have passed).